moabb/datasets/liu2024.py

```diff
@@ -0,0 +1,310 @@
+            path_folder / "edffile" / sub / "eeg" / f"{sub}_task-motor-imagery_eeg.edf"
+        )
+        subject_paths.append(str(subject_path))
+
+        return subject_paths
+
+    @staticmethod
+    def encoding(events_df: pd.DataFrame) -> Tuple[np.array, Dict[int, str]]:
+        """Encode the columns 'value' and 'trial_type' into a single event type.
+
+        Parameters
+        ----------
+        events_df : pd.DataFrame
+            DataFrame containing the events information.
+
+        Returns
+        -------
+        np.ndarray
+            Array of encoded event types.
+
+        Notes
+        -----
+        The 'trial_type' variable can take the following values:
+         - 1 : Left hand
+         - 2 : Right hand
+
+        The 'value' variable can take the following values:
+         - 1 : instructions
+         - 2 : MI
+         - 3 : break
+
+        """
+        # Define the mapping dictionary
+        encoding_mapping = {
+            (2, 2): 1,  # Left hand, MI
+            (1, 2): 2,  # Right hand, MI
+            (1, 1): 3,  # Right hand, instructions
+            (1, 3): 4,  # Right hand, break
+            (2, 1): 3,  # Left hand, instructions
+            (2, 3): 4,  # Left hand, break
+        }
+
+        mapping = {
+            0: "left_hand",
+            1: "right_hand",
+            2: "instr",
+            3: "break",
+        }
+        # Apply the mapping to the DataFrame
+        event_category = events_df.apply(
+            lambda row: encoding_mapping[(row["trial_type"], row["value"])], axis=1
+        )
+
+        return event_category, mapping
+
+    def _get_single_subject_data(self, subject):
+        """Return the data of a single subject.
+
+        Parameters
+        ----------
+        subject : int
+            The subject number to fetch data for.
+
+        Returns
+        -------
+        dict
+            A dictionary containing the raw data for the subject.
+        """
+
+        file_path_list = self.data_path(subject)[0]
+        path_electrodes, path_events = self.data_infos()
+
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            # Read the subject's raw data
+            raw = mne.io.read_raw_edf(
+                file_path_list, verbose=False, infer_types=True, stim_channel=""
+            )
+        # Normalize and Read the montage
+        path_electrodes = self._normalize_extension(path_electrodes)
+        # Read and set the montage
+        montage = read_custom_montage(path_electrodes)
+
+        events_df = pd.read_csv(path_events, sep="\t")
+
+        # Encode the events
+        event_category, mapping = self.encoding(events_df=events_df)
+
+        events = self.create_event_array(raw=raw, event_category=event_category)
+
+        # Creating and setting annotations from the events
+        annotations = mne.annotations_from_events(
+            events, sfreq=raw.info["sfreq"], event_desc=mapping
+        )
```

Comment:
I loaded data of subject 1 and got the following annotation:
```python
>>> data = Liu2024().get_data([1])
>>> data[1]['0']['0'].annotations
 <Annotations | 80 segments: break (40), instr (20), right_hand (20)>
```
Three questions:
1. Is it normal to only have no `left_hand` events?
2. Why are there only 20 `right_hand` events while the docstring states there should be 40 events per class?
3. Why is there two breaks per imagery event?

Reply:
1- The encoding wasn't right, it's solved now.
2-  You're right it should be 20, i counted the whole Imagery events which are 40 ( left + right) 
3- Because both breaks for left_hand and right_hand are encoded as break so both instruction and break are 40 and right_hand, left_hand are 20 each. It should be good now with the last push.